store/src/java/com/zimbra/cs/service/admin/GetDomainInfo.java

```diff
@@ -129,8 +133,13 @@ private void addAttrElementIfNotNull(Entry entry, Element element, String key) t
                 element.addNonUniqueElement(AdminConstants.E_A).addAttribute(AdminConstants.A_N, key).setText(value);
             }
         } else {
+            // checking whether Modern UI is installed
+            File modernUIFile = new File(modernUIFolderPath);
             String value = entry.getAttr(key, null);
-            if (value != null) {
+            if (key.equals(ZAttrProvisioning.A_zimbraModernWebClientDisabled) && !(modernUIFile.exists() && modernUIFile.isDirectory())) {
```

Comment:
Is the condition correct?
* If zimbraModernWebClientDisabled is TRUE, modern client is disabled. It is okay to return TRUE as it is.
* If zimbraModernWebClientDisabled is FALSE, modern client is enabled. However, if the modern directory does not exist, Modern UI is disabled. (i.e. zimbraModernWebClientDisabled should be set to TRUE)
* If zimbraModernWebClientDisabled is empty, process in the same way as FALSE.

Reply:
`key.equals(ZAttrProvisioning.A_zimbraModernWebClientDisabled)` is just checking whether its the attribute on which we want to add a check for

Reply:
Oh, sorry it was my misunderstanding. I though it checked a value. Thanks @dasiyogesh